src/back/loader/Base.js

```diff
@@ -72,7 +72,9 @@ BaseLoader.prototype.normalizeStylesheet = function (style) {
 
 BaseLoader.prototype.load = function () {
     this.mml = this.loadFile();
-    this.postprocess();
+    if (this.project.config.parsed_opts.renderer == 'carto') {
```

Comment:
Magnacarto and Carto expect the styles in slightly different format. While Carto expects an object of the form { id: x, data: y } Magnacarto expects just a string.